cmd/completion.go

```diff
@@ -0,0 +1,224 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+var shell string
+
+func MakeCompletion() *cobra.Command {
+	var completionCmd = &cobra.Command{
+		Use:   "completion SHELL",
+		Short: "Generates shell completion",
+		Long: `Generates shell auto completion for Bash or ZSH.
+		
+Please follow the instructions in the link below to activate the shell auto completion in your environment:
+https://docs.openfaas.com/cli/completion/`,
+		RunE: runCompletion,
+	}
+
+	completionCmd.Flags().StringVar(&shell, "shell", "", "Outputs shell completion, must be bash or zsh")
+	completionCmd.MarkFlagRequired("shell")
+
+	return completionCmd
+}
+
+func runCompletion(cmd *cobra.Command, args []string) (err error) {
+	if shell == "" {
```

Comment:
This can be done with the default case of the switch statement.

Reply:
kept this validation, but its now used to read from the environment when shell argument is missing.